icu4c/source/common/rbbi_cache.cpp

```diff
@@ -158,11 +159,12 @@ void RuleBasedBreakIterator::DictionaryCache::populateDictionary(int32_t startPo
 
         // We now have a dictionary character. Get the appropriate language object
         // to deal with it.
-        const LanguageBreakEngine *lbe = fBI->getLanguageBreakEngine(c);
+        LanguageBreakEngine *lbe = fBI->getLanguageBreakEngine(c);
 
         // Ask the language object if there are any breaks. It will add them to the cache and
         // leave the text pointer on the other side of its range, ready to search for the next one.
         if (lbe != NULL) {
+            lbe->enablePhraseBreaking(fBI->fIsPhraseBreaking);
```

Comment:
you should NOT change the LanguageBreakEngine after you create it. The LanguageBreakEngine is shared and used by multiple threads and multiple RBBIs currently, if you change it's internal state, the usage of it from a different thread will then be changes. you could choose to either
1. change the  getLanguageBreakEngine() to pass in fBI->fIsPhraseBreaking so it will return a different one OR
2. change the LanguageBreakEngine interface to somehow pass in the fBI->fIsPhraseBreaking as information in the breaking code (but not remembered in the internal state of LanguageBreakEngine itself)